official/recommendation/ncf_main.py

```diff
@@ -201,100 +210,135 @@ def main(_):
     tf.logging.info("Starting a training cycle: {}/{}".format(
-        cycle_index, total_training_cycle - 1))
+        cycle_index + 1, total_training_cycle))
 
     # Train the model
-    train_cycle_begin = time.time()
-    estimator.train(input_fn=train_input_fn,
-                    hooks=[tf.train.ProfilerHook(save_steps=10000)])
-    train_cycle_end = time.time()
+    estimator.train(input_fn=train_input_fn, hooks=train_hooks)
 
     # Evaluate the model
-    eval_cycle_begin = time.time()
-    hr, ndcg = evaluate_model(
-        estimator, FLAGS.batch_size, num_gpus, ncf_dataset.eval_true_items,
-        ncf_dataset.eval_all_items, FLAGS.num_parallel_calls)
-    eval_cycle_end = time.time()
-
-    # Log the train time, evaluation time, and HR and NDCG results.
+    eval_results = evaluate_model(
+        estimator, FLAGS.batch_size, num_gpus, ncf_dataset)
+
+    # Benchmark the evaluation results
+    benchmark_logger.log_evaluation_result(eval_results)
+    # Log the HR and NDCG results.
+    hr = eval_results[_HR_KEY]
+    ndcg = eval_results[_NDCG_KEY]
     tf.logging.info(
-        "Iteration {} [{:.1f} s]: HR = {:.4f}, NDCG = {:.4f}, [{:.1f} s]"
-        .format(cycle_index, train_cycle_end - train_cycle_begin, hr, ndcg,
-                eval_cycle_end - eval_cycle_begin))
-
-  # Remove temporary files
-  os.remove(constants.TRAIN_DATA)
-  os.remove(constants.TEST_DATA)
+        "Iteration {}: HR = {:.4f}, NDCG = {:.4f}".format(
+            cycle_index + 1, hr, ndcg))
+
+    # If some evaluation threshold is met
+    if model_helpers.past_stop_threshold(FLAGS.hr_threshold, hr):
+      break
+
+  # Clear the session explicitly to avoid session delete error
+  tf.keras.backend.clear_session()
+
+
+def define_ncf_flags():
+  """Add flags for running ncf_main."""
+  # Add common flags
+  flags_core.define_base(export_dir=False)
+  flags_core.define_performance(
+      num_parallel_calls=False,
+      inter_op=False,
+      intra_op=False,
+      synthetic_data=False,
+      max_train_steps=False,
+      dtype=False
+  )
+  flags_core.define_benchmark()
+
+  flags.adopt_module_key_flags(flags_core)
+
+  flags_core.set_defaults(
+      model_dir="/tmp/ncf/",
+      data_dir="/tmp/movielens-data/",
+      train_epochs=2,
+      batch_size=256,
+      hooks="ProfilerHook")
+
+  # Add ncf-specific flags
+  flags.DEFINE_enum(
+      name="dataset", short_name="ds", default="ml-1m",
+      enum_values=["ml-1m", "ml-20m"], case_sensitive=False,
+      help=flags_core.help_wrap(
+          "Dataset to be trained and evaluated. Two datasets are "
+          "available for now: ml-1m and ml-20m."))
+
+  flags.DEFINE_integer(
+      name="num_factors", short_name="nf", default=8,
+      help=flags_core.help_wrap("The Embedding size of MF model."))
+
+  flags.DEFINE_list(
+      name="layers", short_name="ly", default=[64, 32, 16, 8],
+      help=flags_core.help_wrap("The size of hidden layers for MLP."))
```

Comment:
This should be a list? How does one enter the list? Comma-separated ints? Please define expectations for the user, display default.